src/Raven.Client/Documents/Session/AsyncDocumentQuery.cs

```diff
@@ -771,7 +771,14 @@ Lazy<Task<int>> IAsyncDocumentQueryBase<T>.CountLazilyAsync(CancellationToken to
         }
 
         /// <inheritdoc />
-        Task<List<T>> IAsyncDocumentQueryBase<T>.ToListAsync(CancellationToken token)
+        async Task<List<T>> IAsyncDocumentQueryBase<T>.ToListAsync(CancellationToken token)
+        {
+            var items = await ExecuteQueryOperation(null, token).ConfigureAwait(false);
```

Comment:
Just invoke previous implementation and then do `ToArray` on the results as a shortcut.